src/codegen/generators/expression_generator.rs

```diff
@@ -15,7 +15,7 @@ pub struct ExpressionCodeGenerator<'a, 'b> {
     /// an optional type hint for generating literals
     type_hint: Option<DataTypeInformation>,
     /// the current function to create blocks in
-    function_context: Option<&'b FunctionContext<'a>>,
+    function_context: Option<&'a FunctionContext<'a>>,
```

Comment:
its not necessary.
I moved it back to 'b (in all locations) and it compiles?

you essentially say: I gonna borrow the functionContext which is going to live 'a long and I keep the borrow for its whole lifetime. It doesnt look like it's necessary.

Reply:
you're right, for some reason i had issues with it at first, but i guess I got them solved later without realizing it.